### PR TITLE
feat: add ship-type-specific URLs for better unfurling

### PR DESCRIPTION
**Pull request by steven-noorbergen**

### CODE REVIEW

File: app/[shipname]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      ({
+        shipname: name.toLowerCase(),
+      }) as PageParams,
+  );
+}
+
+export async function generateMetadata({ params: { shipname } }: { params: PageParams }): Promise<Metadata> {
+  const currentShipType = await getCurrentShipType(shipname);
+
+  if (!currentShipType) return {};
```
Comment (TrueBrain) [inline app/[shipname]/page.tsx:44]:
```suggestion
  if (currentShipType === undefined) return {};
```

**Comment by TrueBrain** [comment]
As discussed on Discord, in the end I decided to go for a full-out approach to this problem, and #75 renders the complete fit in an OpenGraph image.

Mostly the reasoning for going the full route immediately, is that I was a bit afraid that otherwise we could get stuck on URLs with shiptypes in it, which are a bit annoying to maintain (as they need to be done compile-time). With #75 there is no need, as the rendering will tell you exactly what ship it is.

Either way, I used a lot of snippets from this PR to build up #75, and more than one that came before. This PR showed the need to have all the Providers in the Layouter, which is now the case in `main` already.

In short: tnx for this work :) It wasn't in vain :)
